internal/resource/redirectvalidation/redirect_validation_resource.go

```diff
@@ -0,0 +1,417 @@
+			addRequest.RedirectValidationLocalSettings.EnableInErrorResourceValidation = internaltypes.AttrValueToBoolPointer(redirectValidationLocalSettingsAttrs["enable_in_error_resource_validation"])
+		}
+
+		whiteListPlan := plan.RedirectValidationLocalSettings.Attributes()["white_list"].(types.Set)
+		if internaltypes.IsDefined(whiteListPlan) {
+			addRequest.RedirectValidationLocalSettings.WhiteList = []client.RedirectValidationSettingsWhitelistEntry{}
+			for i := 0; i < len(whiteListPlan.Elements()); i++ {
+				item := whiteListPlan.Elements()[i].(types.Object)
+				itemAttrs := item.Attributes()
+				newWhiteList := client.NewRedirectValidationSettingsWhitelistEntryWithDefaults()
+				newWhiteList.SetTargetResourceSSO(itemAttrs["target_resource_sso"].(types.Bool).ValueBool())
+				newWhiteList.SetTargetResourceSLO(itemAttrs["target_resource_slo"].(types.Bool).ValueBool())
+				newWhiteList.SetInErrorResource(itemAttrs["in_error_resource"].(types.Bool).ValueBool())
+				newWhiteList.SetIdpDiscovery(itemAttrs["idp_discovery"].(types.Bool).ValueBool())
+				newWhiteList.SetValidDomain(itemAttrs["valid_domain"].(types.String).ValueString())
+				newWhiteList.SetValidPath(itemAttrs["valid_path"].(types.String).ValueString())
+				newWhiteList.SetAllowQueryAndFragment(itemAttrs["allow_query_and_fragment"].(types.Bool).ValueBool())
+				newWhiteList.SetRequireHttps(itemAttrs["require_https"].(types.Bool).ValueBool())
+				addRequest.RedirectValidationLocalSettings.WhiteList = append(addRequest.RedirectValidationLocalSettings.WhiteList, *newWhiteList)
+			}
+		}
+	}
+
+	if internaltypes.IsDefined(plan.RedirectValidationPartnerSettings) {
+		addRequest.RedirectValidationPartnerSettings = client.NewRedirectValidationPartnerSettings()
+		enableWreplyValidationSloAttrs := plan.RedirectValidationPartnerSettings.Attributes()["enable_wreply_validation_slo"]
+		if internaltypes.IsDefined(enableWreplyValidationSloAttrs) {
+			enableWreplyValidationSlo := internaltypes.ConvertToPrimitive(enableWreplyValidationSloAttrs).(bool)
+			addRequest.RedirectValidationPartnerSettings.EnableWreplyValidationSLO = &enableWreplyValidationSlo
+		}
+	}
+
+	return nil
+
+}
+
+// Metadata returns the resource type name.
+func (r *redirectValidationResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_redirect_validation"
+}
+
+func (r *redirectValidationResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func readRedirectValidationResponse(ctx context.Context, r *client.RedirectValidationSettings, state *redirectValidationResourceModel, expectedValues *redirectValidationResourceModel) {
```

Comment:
Looks like expectedValues isn't needed in this method

Reply:
Fixed in the latest commit